homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,161 @@
+            self._attr_temperature_unit = TEMP_CELSIUS
+            self._attr_precision = PRECISION_HALVES
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current HVAC mode."""
+        mode = self._client.get_heatmode()
+        return self._balboa_to_ha_heatmode_map[mode]
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current operation mode."""
+        state = self._client.get_heatstate()
+        if state >= self._client.ON:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def fan_mode(self) -> str:
+        """Return the current fan mode."""
+        fanmode = self._client.get_blower()
+        return self._balboa_to_ha_blower_map.get(fanmode, FAN_OFF)
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._client.get_curtemp()
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature we try to reach."""
+        return self._client.get_settemp()
+
+    @property
+    def preset_mode(self):
+        """Return current preset mode."""
+        return self._client.get_heatmode(True)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set a new target temperature."""
+        scale = self._client.get_tempscale()
+        newtemp = kwargs[ATTR_TEMPERATURE]
+        if newtemp > self._client.tmax[self._client.TEMPRANGE_LOW][scale]:
+            await self._client.change_temprange(self._client.TEMPRANGE_HIGH)
+            await asyncio.sleep(1)
+        if newtemp < self._client.tmin[self._client.TEMPRANGE_HIGH][scale]:
+            await self._client.change_temprange(self._client.TEMPRANGE_LOW)
+            await asyncio.sleep(1)
+        await self._client.send_temp_change(newtemp)
+
+    async def async_set_preset_mode(self, preset_mode) -> None:
+        """Set new preset mode."""
+        modelist = self._client.get_heatmode_stringlist()
+        self._async_validate_mode_or_raise(preset_mode)
+        if preset_mode not in modelist:
+            raise HomeAssistantError(f"{preset_mode} is not a valid preset mode")
+        await self._client.change_heatmode(modelist.index(preset_mode))
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new fan mode."""
+        await self._client.change_blower(self._ha_to_balboa_blower_map[fan_mode])
+
+    def _async_validate_mode_or_raise(self, mode):
+        """Check that the mode can be set."""
+        if mode == self._client.HEATMODE_RNR:
+            raise HomeAssistantError(f"{mode} can only be reported but not set")
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode.
+
+        OFF = Rest
+        AUTO = Ready in Rest (can't be set, only reported)
+        HEAT = Ready
+        """
+        mode = self._ha_heatmode_to_balboa_map[hvac_mode]
+        self._async_validate_mode_or_raise(mode)
```

Comment:
Why do we allow `HVAC_MODE_AUTO` in the list of modes if we don't support to set it? That list is for setting.